src/containers/Tenant/Query/QueryEditor/QueryEditor.tsx

```diff
@@ -173,6 +174,7 @@ export default function QueryEditor(props: QueryEditorProps) {
         if (!partial) {
             if (text !== historyQueries[historyCurrentIndex]?.queryText) {
                 dispatch(saveQueryToHistory({queryText: text, queryId}));
+                dispatch(setIsDirty(false));
```

Comment:
If I run some query twice, for the second run it won't be saved in history again and confirmation will be shown. This row should be `if(!partial)` block